diff --git a/src/index.js b/src/index.js
index e9db44b9fec73767c840beb25c400030c191de44..1f740e8074f76afc7c7e39ebe0a07371eea24b3c 100644
--- a/src/index.js
+++ b/src/index.js
@@ -90,7 +90,7 @@ function advancedPreset(opts = {}) {
           [postcssDiscardUnused, 'discardUnused'],
           [postcssMergeIdents, 'mergeIdents'],
           [postcssReduceIdents, 'reduceIdents'],
-          [postcssZindex, 'zindex'],
+          // [postcssZindex, 'zindex'],
         ],
         opts
       ),
