diff --git a/lib/core.js b/lib/core.js
index f59e614f2e8569ce6ffbac9fd7a02b02ee393092..3b695b2a590296c80e33d10c4dfee2249c4bcb0f 100644
--- a/lib/core.js
+++ b/lib/core.js
@@ -236,7 +236,7 @@ function stringify(value, language) {
     type: value.type,
     content: stringify(value.content, language),
     tag: 'span',
-    classes: ['token', value.type],
+    classes: [value.type],
     attributes: {},
     language
   }
