diff --git a/index.js b/index.js
index 18810b1a4c778a2ef2726f9fc54b786a7fb65b06..caf5ceff4a3ed5072c75d72faac5f2395bf09447 100644
--- a/index.js
+++ b/index.js
@@ -6,9 +6,10 @@
  * @returns {Array<string>}
  *   List of tokens.
  */
+const Space_Regx = /[ \t\n\r\f]+/g
 export function parse(value) {
   const input = String(value || '').trim()
-  return input ? input.split(/[ \t\n\r\f]+/g) : []
+  return input ? input.split(Space_Regx) : []
 }
 
 /**
